File: example-base-28/nodejs-utilities-playground-v2/src/dsa/arrays/two-sum.js
function twoSum(numbers, target) {
  const seen = new Map();
  for (let i = 0; i < numbers.length; i++) {
    const complement = target - numbers[i];
    if (seen.has(complement)) return [seen.get(complement), i];
    seen.set(numbers[i], i);
  }
  return [];
}
module.exports = { twoSum };

File: example-base-29/python-algorithms-playground/algorithms/arrays/two_sum.py
def two_sum(nums, target):
    seen = {}
    for i, value in enumerate(nums):
        complement = target - value
        if complement in seen:
            return [seen[complement], i]
        seen[value] = i
    return []

if __name__ == "__main__":
    print(two_sum([2, 7, 11, 15], 9))
